hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/common/detailslayout/AbstractDistributionSetDetails.java

```diff
@@ -0,0 +1,193 @@
+                managementUIState, tagManagement, distributionSetManagement);
+
+        dsMetadataTable = new DistributionSetMetadatadetailslayout(i18n, permissionChecker, distributionSetManagement,
+                dsMetadataPopupLayout, entityFactory, uiNotification);
+    }
+
+    @Override
+    protected void onEdit(final ClickEvent event) {
+        final Window newDistWindow = distributionAddUpdateWindowLayout.getWindow(getSelectedBaseEntityId());
+        newDistWindow.setCaption(getI18n().get(UIComponentIdProvider.DIST_UPDATE_CAPTION));
+        UI.getCurrent().addWindow(newDistWindow);
+        newDistWindow.setVisible(Boolean.TRUE);
+    }
+
+    @EventBusListenerMethod(scope = EventScope.UI)
+    void onEvent(final DistributionTableEvent distributionTableEvent) {
+        onBaseEntityEvent(distributionTableEvent);
+    }
+
+    @Override
+    protected String getDefaultCaption() {
+        return getI18n().get("distribution.details.header");
+    }
+
+    @Override
+    protected String getEditButtonId() {
+        return UIComponentIdProvider.DS_EDIT_BUTTON;
+    }
+
+    @Override
+    protected Boolean hasEditPermission() {
+        return getPermissionChecker().hasUpdateDistributionPermission();
+    }
+
+    @Override
+    protected String getTabSheetId() {
+        return UIComponentIdProvider.DISTRIBUTIONSET_DETAILS_TABSHEET_ID;
+    }
+
+    @Override
+    protected void populateMetadataDetails() {
+        dsMetadataTable.populateDSMetadata(getSelectedBaseEntity());
+    }
+
+    @Override
+    protected String getDetailsHeaderCaptionId() {
+        return UIComponentIdProvider.DISTRIBUTION_DETAILS_HEADER_LABEL_ID;
+    }
+
+    @Override
+    protected Boolean isMetadataIconToBeDisplayed() {
+        return true;
+    }
+
+    @Override
+    protected void showMetadata(final ClickEvent event) {
+        final Optional<DistributionSet> ds = distributionSetManagement
+                .findDistributionSetById(getSelectedBaseEntityId());
+        if (!ds.isPresent()) {
+            uiNotification.displayWarning(getI18n().get("distributionset.not.exists"));
+            return;
+        }
+        UI.getCurrent().addWindow(dsMetadataPopupLayout.getWindow(ds.get(), null));
+    }
+
+    @Override
+    protected void addTabs(final TabSheet detailsTab) {
+        detailsTab.addTab(createDetailsLayout(), getI18n().get("caption.tab.details"), null);
+        detailsTab.addTab(createDescriptionLayout(), getI18n().get("caption.tab.description"), null);
+        detailsTab.addTab(createSoftwareModuleTab(), getI18n().get("caption.softwares.distdetail.tab"), null);
+        detailsTab.addTab(createTagsLayout(), getI18n().get("caption.tags.tab"), null);
+        detailsTab.addTab(createLogLayout(), getI18n().get("caption.logs.tab"), null);
+        detailsTab.addTab(dsMetadataTable, getI18n().get("caption.metadata"), null);
+    }
+
+    protected VerticalLayout createTagsLayout() {
```

Comment:
Because I add the tagsLayout to the detailsTab. 
detailsTab.addTab(createTagsLayout(), getI18n().get("caption.tags.tab"), null);